### PR TITLE
x-nullable $ref fix

### PR DESCRIPTION
**Pull request by hmajid2301**
Fixes # 1070.

Changes proposed in this pull request:

 - Allow x-nullable to follow $ref (resolve) for v1

### CODE REVIEW

File: connexion/decorators/validation.py [hunk 1]
```diff
@@ -89,6 +89,11 @@ def extend_with_nullable_support(validator_class):
     def nullable_support(validator, properties, instance, schema):
         null_properties = {}
         for property_, subschema in six.iteritems(properties):
+            if "$ref" in subschema:
+                ref_schema = validator.resolver.resolve(subschema["$ref"])
+                if len(ref_schema) == 2:
```
Comment (dtkav) [inline connexion/decorators/validation.py]: what does this do?
can you remove the magic number / add a comment?

**Comment by dtkav** [comment]
@hjacobs - what did you have in mind for the v1_unsupported branch?
Should it still pass CI? Same standards for code review?

**Comment by hjacobs** [comment]
@dtkav I don't want to support v1, but at least we can have a branch for it so bugfixes have a "home" (and others might be able to look at them). Haven't decided on releases yet.

File: connexion/decorators/validation.py (hunk 1 above)
Comment (hmajid2301) [reply connexion/decorators/validation.py]: `ref_schema` returns a tuple like so like the $ref and the actual schema definition ("definitions/foo", {}). Also yeh I can do that if you want.

Comment (dtkav) [reply connexion/decorators/validation.py]: I had done the review before realizing this was against v1_unsupported.
I don't have a strong preference.

Another way:
```
try:
    _, subschema = validator.resolver.resolve(subschema["$ref"])
except ValueError:
    pass
```

**Comment by dtkav** [comment]
Ok, that makes sense. Then I suppose if this is a sort of "user contrib" branch, then we probably don't need to have the same standards of code review and passing CI.

It seems unlikely that we'll be able to get this to pass CI without a lot of work anyway, because it's probably still testing deprecated python versions, and package versions aren't pinned to known working versions.

Are you happy to merge this then?

File: connexion/decorators/validation.py (hunk 1 above)
Comment (hmajid2301) [reply connexion/decorators/validation.py]: Yes that is much cleaner

**Comment by hmajid2301** [comment]
What's the best to install the latest v1 version then with these changes ?

**Comment by hjacobs** [comment]
@hmajid2301 you can install directly from the git branch:

```
pip install git+https://github.com/zalando/connexion.git@v1_unsupported
```

**Pull request merged**